js/exporter/image_creator.js

```diff
@@ -419,31 +419,44 @@ function drawElement(element, context, parentOptions, shared) {
 
     if(!isText) {
         applyFilter(context, options, shared);
-        fillElement(context, options, shared);
+        if(!isImage) {
+            promise = fillElement(context, options, shared);
+        }
         strokeElement(context, options);
     }
 
-    applyGradient(context, options, shared, element);
+    applyGradient(context, options, shared, element, 'linear');
+    applyGradient(context, options, shared, element, 'radial');
 
     context.restore();
 
     return promise;
 }
 
-function applyGradient(context, options, { gradients }, element) {
-    if(gradients.length === 0) {
+function applyGradient(context, options, { linearGradients, radialGradients }, element, type) {
```

Comment:
did I understand correctly that the function is not pure?
since we are changing contexts ?

Reply:
Yes, canvas mutation difficult made with pure functions.

Reply:
Since it has a change effect on the object, maybe it's worth putting it inside the object, or is it currently impossible to do it in a simple way ?